hudi-common/src/main/java/org/apache/hudi/keygen/KeyGenerator.java

```diff
@@ -56,4 +61,35 @@ public List<String> getRecordKeyFieldNames() {
     throw new UnsupportedOperationException("Bootstrap not supported for key generator. "
         + "Please override this method in your custom key generator.");
   }
+
+  public static String constructRecordKey(List<String> recordKeyFields, List<Object> recordKeyValues) {
```

Comment:
nit: change this to take `List<Pair<String, Object>> recordKeyFieldAndValueList` so we don't have to worry that the list size does not match or invocation with mismatched list?

Reply:
Another consideration is to pass in the function to generate the record key value so there is no need to construct a list of record key values, given this is invoked at per-record level.